packages/babel-plugin-strip-runtime/src/index.ts

```diff
@@ -47,6 +51,44 @@ export default declare<PluginPass>((api) => {
               // If we used ESM it would blow up with CJS source, unfortunately.
             });
           }
+
+          if (this.opts.extractStylesToDirectory && this.styleRules.length > 0) {
+            // Build filename of the css file
+            const currentFilename = basename(filename);
+            const cssFilename = `${currentFilename.substring(
+              0,
+              currentFilename.lastIndexOf('.')
+            )}.compiled.css`;
+
+            if (!file.opts.generatorOpts?.sourceFileName) {
```

Comment:
`souceFileName` seems returning the baseName. Are you meant to use `filenameRelative`?

Reply:
I use `sourceFileName` as that specifies the source file path based on where we are

For example, we're transforming `src/my-file.tsx` which will go to `dist/my-file.tsx`. `sourceFileName` should be `../src/my-file.tsx`.

Does that all make sense? I think I can leave some comments to make that clearer, as this was the bulk of the complexity (juggling the output file path)

Reply:
Ah I see `filenameRelative` is in the implementation but not the types, I'll see if I can use that 💯

Reply:
Looking at https://babeljs.io/docs/options#filenamerelative, it seems `sourceFilename` is the variable we get from that configuration option